### PR TITLE
Sonar fixes.

### PR DESCRIPTION
**Pull request by prasa7**

### CODE REVIEW

**Review by shilmyhasan — APPROVED** [review]

File: components/apimgt/org.wso2.carbon.apimgt.persistence/src/main/java/org/wso2/carbon/apimgt/persistence/RegistryPersistenceImpl.java [hunk 1]
```diff
@@ -1721,8 +1715,9 @@ public void saveWSDL(Organization org, String apiId, ResourceFile wsdlResourceFi
             String visibility = apiArtifact.getAttribute(APIConstants.API_OVERVIEW_VISIBILITY);
             String visibleRolesList = apiArtifact.getAttribute(APIConstants.API_OVERVIEW_VISIBLE_ROLES);
 
-            Resource wsdlResource = registry.newResource();;
-            
+            Resource wsdlResource = registry.newResource();
+            ;
```
Comment (tharindu1st) [inline components/apimgt/org.wso2.carbon.apimgt.persistence/src/main/java/org/wso2/carbon/apimgt/persistence/RegistryPersistenceImpl.java:1719]: remove ;

File: components/apimgt/org.wso2.carbon.apimgt.persistence/src/main/java/org/wso2/carbon/apimgt/persistence/RegistryPersistenceImpl.java [hunk 2]
```diff
@@ -1838,14 +1833,15 @@ public ResourceFile getWSDL(Organization org, String apiId) throws WSDLPersisten
             }
         }
     }
+
```
Comment (tharindu1st) [inline components/apimgt/org.wso2.carbon.apimgt.persistence/src/main/java/org/wso2/carbon/apimgt/persistence/RegistryPersistenceImpl.java:1836]: new line ?

File: components/apimgt/org.wso2.carbon.apimgt.persistence/src/main/java/org/wso2/carbon/apimgt/persistence/RegistryPersistenceImpl.java [hunk 3]
```diff
@@ -3511,16 +3510,17 @@ public void deleteAPIProduct(Organization org, String apiId) throws APIPersisten
                 RegistryPersistenceUtil.endTenantFlow();
             }
         }
-    
+
     }
+
```
Comment (tharindu1st) [inline components/apimgt/org.wso2.carbon.apimgt.persistence/src/main/java/org/wso2/carbon/apimgt/persistence/RegistryPersistenceImpl.java:3515]: new line

File: components/apimgt/org.wso2.carbon.apimgt.persistence/src/main/java/org/wso2/carbon/apimgt/persistence/RegistryPersistenceImpl.java [hunk 4]
```diff
@@ -3402,12 +3402,11 @@ public PublisherAPIProduct updateAPIProduct(Organization org, PublisherAPIProduc
...
         } finally {
             try {
-                if (!transactionCommitted) {
+                if (!transactionCommitted && registry != null) {
                     registry.rollbackTransaction();
                 }
-
             } catch (RegistryException ex) {
-                throw new APIPersistenceException("Error occurred while rolling back the transaction.", ex);
+                log.error("Error occurred while rolling back the transaction.");
```
Comment (tharindu1st) [inline components/apimgt/org.wso2.carbon.apimgt.persistence/src/main/java/org/wso2/carbon/apimgt/persistence/RegistryPersistenceImpl.java:3409]: add ex as well

File: components/apimgt/org.wso2.carbon.apimgt.persistence/src/main/java/org/wso2/carbon/apimgt/persistence/RegistryPersistenceImpl.java [hunk 5]
```diff
@@ -3402,12 +3402,11 @@ public PublisherAPIProduct updateAPIProduct(Organization org, PublisherAPIProduc
             throw new APIPersistenceException("Error while performing registry transaction operation", e);
         } finally {
             try {
-                if (!transactionCommitted) {
+                if (!transactionCommitted && registry != null) {
```
Comment (tharindu1st) [inline components/apimgt/org.wso2.carbon.apimgt.persistence/src/main/java/org/wso2/carbon/apimgt/persistence/RegistryPersistenceImpl.java:3405]: registry != null && !transactionCommitted

File: components/apimgt/org.wso2.carbon.apimgt.persistence/src/main/java/org/wso2/carbon/apimgt/persistence/RegistryPersistenceImpl.java [hunk 6]
```diff
@@ -3195,12 +3195,12 @@ public PublisherAPIProduct addAPIProduct(Organization org, PublisherAPIProduct p
...
                     registry.rollbackTransaction();
                 }
             } catch (RegistryException ex) {
-                throw new APIPersistenceException("Error while rolling back the transaction for API Product : "
-                        + publisherAPIProduct.getApiProductName(), ex);
+                log.error("Error while rolling back the transaction for API Product : "
```
Comment (tharindu1st) [inline components/apimgt/org.wso2.carbon.apimgt.persistence/src/main/java/org/wso2/carbon/apimgt/persistence/RegistryPersistenceImpl.java:3202]: add ex as well

File: components/apimgt/org.wso2.carbon.apimgt.persistence/src/main/java/org/wso2/carbon/apimgt/persistence/RegistryPersistenceImpl.java [hunk 7]
```diff
@@ -3195,12 +3195,12 @@ public PublisherAPIProduct addAPIProduct(Organization org, PublisherAPIProduct p
             throw new APIPersistenceException("Error while creating API Product", e);
         } finally {
             try {
-                if (!transactionCommitted) {
+                if (!transactionCommitted && registry != null) {
```
Comment (tharindu1st) [inline components/apimgt/org.wso2.carbon.apimgt.persistence/src/main/java/org/wso2/carbon/apimgt/persistence/RegistryPersistenceImpl.java:3198]: registry != null && !transactionCommitted

File: components/apimgt/org.wso2.carbon.apimgt.persistence/src/main/java/org/wso2/carbon/apimgt/persistence/RegistryPersistenceImpl.java [hunk 8]
```diff
@@ -849,19 +843,19 @@ public void deleteAPI(Organization org, String apiId) throws APIPersistenceExcep
...
+            transactionCommitted = true;
         } catch (RegistryException e) {
             throw new APIPersistenceException("Failed to remove the API : " + apiId, e);
         } finally {
             if (tenantFlowStarted) {
                 RegistryPersistenceUtil.endTenantFlow();
             }
             try {
-                if (!transactionCommitted) {
+                if (!transactionCommitted && registry != null) {
```
Comment (tharindu1st) [inline components/apimgt/org.wso2.carbon.apimgt.persistence/src/main/java/org/wso2/carbon/apimgt/persistence/RegistryPersistenceImpl.java:854]: registry != null && !transactionCommitted

File: components/apimgt/org.wso2.carbon.apimgt.persistence/src/main/java/org/wso2/carbon/apimgt/persistence/RegistryPersistenceImpl.java [hunk 9]
```diff
@@ -631,11 +625,11 @@ public PublisherAPI updateAPI(Organization org, PublisherAPI publisherAPI) throw
...
```
Comment (tharindu1st) [inline components/apimgt/org.wso2.carbon.apimgt.persistence/src/main/java/org/wso2/carbon/apimgt/persistence/RegistryPersistenceImpl.java:628]: registry != null && !transactionCommitted

File: components/apimgt/org.wso2.carbon.apimgt.persistence/src/main/java/org/wso2/carbon/apimgt/persistence/RegistryPersistenceImpl.java [hunk 10]
```diff
@@ -631,11 +625,11 @@ public PublisherAPI updateAPI(Organization org, PublisherAPI publisherAPI) throw
...
                     registry.rollbackTransaction();
                 }
             } catch (RegistryException ex) {
-                throw new APIPersistenceException("Error occurred while rolling back the transaction. ", ex);
+                log.error("Error occurred while rolling back the transaction. ");
```
Comment (tharindu1st) [inline components/apimgt/org.wso2.carbon.apimgt.persistence/src/main/java/org/wso2/carbon/apimgt/persistence/RegistryPersistenceImpl.java:632]: add ex

**Review by tharindu1st — APPROVED** [review]

File: components/apimgt/org.wso2.carbon.apimgt.persistence/src/main/java/org/wso2/carbon/apimgt/persistence/RegistryPersistenceImpl.java (hunk 2 above)
Comment (prasa7) [reply components/apimgt/org.wso2.carbon.apimgt.persistence/src/main/java/org/wso2/carbon/apimgt/persistence/RegistryPersistenceImpl.java:1836]: This is a formatting fix

File: components/apimgt/org.wso2.carbon.apimgt.persistence/src/main/java/org/wso2/carbon/apimgt/persistence/RegistryPersistenceImpl.java (hunk 3 above)
Comment (prasa7) [reply components/apimgt/org.wso2.carbon.apimgt.persistence/src/main/java/org/wso2/carbon/apimgt/persistence/RegistryPersistenceImpl.java:3515]: This is a formatting fix

File: components/apimgt/org.wso2.carbon.apimgt.persistence/src/main/java/org/wso2/carbon/apimgt/persistence/RegistryPersistenceImpl.java (hunk 1 above)
Comment (prasa7) [reply components/apimgt/org.wso2.carbon.apimgt.persistence/src/main/java/org/wso2/carbon/apimgt/persistence/RegistryPersistenceImpl.java]: removed

File: components/apimgt/org.wso2.carbon.apimgt.persistence/src/main/java/org/wso2/carbon/apimgt/persistence/RegistryPersistenceImpl.java (hunk 8 above)
Comment (prasa7) [reply components/apimgt/org.wso2.carbon.apimgt.persistence/src/main/java/org/wso2/carbon/apimgt/persistence/RegistryPersistenceImpl.java]: fixed

File: components/apimgt/org.wso2.carbon.apimgt.persistence/src/main/java/org/wso2/carbon/apimgt/persistence/RegistryPersistenceImpl.java (hunk 6 above)
Comment (prasa7) [reply components/apimgt/org.wso2.carbon.apimgt.persistence/src/main/java/org/wso2/carbon/apimgt/persistence/RegistryPersistenceImpl.java:3201]: added

File: components/apimgt/org.wso2.carbon.apimgt.persistence/src/main/java/org/wso2/carbon/apimgt/persistence/RegistryPersistenceImpl.java (hunk 4 above)
Comment (prasa7) [reply components/apimgt/org.wso2.carbon.apimgt.persistence/src/main/java/org/wso2/carbon/apimgt/persistence/RegistryPersistenceImpl.java]: added

**Pull request merged**